sdk/src/main/kotlin/com/processout/sdk/core/logger/BaseLoggerService.kt

```diff
@@ -0,0 +1,63 @@
+package com.processout.sdk.core.logger
+
+import android.os.Build
+import com.processout.sdk.BuildConfig
+import java.util.Calendar
+
+internal abstract class BaseLoggerService(
+    private val minimumLevel: LogLevel
+) : LoggerService {
+
+    companion object {
+        private const val MAX_TAG_LENGTH_BEFORE_API_26 = 23
+        private const val ATTRIBUTE_LINE = "Line"
+    }
+
+    private val loggerPackageName = LoggerService::class.java.`package`?.name
+
+    override fun log(
+        level: LogLevel,
+        message: String,
+        vararg args: Any?,
+        attributes: Map<String, String>?
+    ) {
+        if (level.priority < minimumLevel.priority)
+            return
+
+        val formattedMessage = if (args.isNotEmpty()) message.format(*args) else message
+
+        val stackTraceElement = Throwable().stackTrace.find { element ->
+            // Find first element in the stack trace that do not belongs to logger package.
+            // This element refers to the class that actually invoked logging.
+            loggerPackageName?.let {
+                element.className.startsWith(it).not()
+            } ?: false
+        }
```

Comment:
Is inspection of stack trace fast enough in Kotlin? Concern here is that we are doing it for every log call so it may cause freezes.

Reply:
Pretty fast

<img width="452" alt="logs_timestamp" src="https://github.com/processout/processout-android/assets/114916876/01b37908-e838-4d5d-8490-9cec57555ceb">